### PR TITLE
Use kubectl to configure route reflector.

### PR DESCRIPTION
**Pull request by sridhartigera**
At present we are using calicoctl to configure route reflectors. Going forward we need to use kubectl as appropriate validations are added to the annotations. 

- For CC and CE, 3.16 
- For OSS, 3.25, 3.24

Issue:

Link to docs preview:

SME review:
- [x] An SME has approved this change. 

DOCS review:
- [x] A member of the docs team has approved this change.

Additional information:

### CODE REVIEW

File: calico-cloud/networking/configuring/bgp.mdx [hunk 1]
```diff
@@ -125,7 +125,7 @@ spec:
 To configure a node to be a route reflector with cluster ID 244.0.0.1, run the following command.
 
 ```
-calicoctl patch node my-node -p '{"spec": {"bgp": {"routeReflectorClusterID": "244.0.0.1"}}}'
+kubectl annotate node my-node projectcalico.org/RouteReflectorClusterID=244.0.0.1
```
Comment (caseydavenport) [inline calico-cloud/networking/configuring/bgp.mdx:128]: This change won't work for etcd mode. We need to use tabs for this - I'm not sure how that works in the new site infra, but @ctauchen might be able to help there.

We should have one tab for "Kubernetes API datastore" mode and one for "etcd datastore" mode, with k8s API being the default tab.

File: calico/networking/configuring/bgp.mdx [hunk 2]
```diff
@@ -119,10 +119,23 @@ spec:
 
 To configure a node to be a route reflector with cluster ID 244.0.0.1, run the following command.
 
+<Tabs>
+<TabItem label="Kubernetes datastore" value="Kubernetes datastore">
+
+```
```
Comment (bmckercher123) [inline calico/networking/configuring/bgp.mdx]: Add ```bash for kubectl and calicoctl throughout

```bash
```

**Comment by ctauchen** [comment]
@sridhartigera You'll need to import the Tabs and TabItem components. Add this before the title of the document: 
```
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
```

File: calico/networking/configuring/bgp.mdx [hunk 3]
```diff
@@ -119,13 +122,26 @@ spec:
 
 To configure a node to be a route reflector with cluster ID 244.0.0.1, run the following command.
 
+<Tabs>
+<TabItem label="Kubernetes datastore" value="Kubernetes datastore-0">
```
Comment (ctauchen) [inline calico/networking/configuring/bgp.mdx]:
```suggestion
<TabItem label="Kubernetes API datastore" value="kubernetes-api-datastore" default>
```

File: calico/networking/configuring/bgp.mdx [hunk 4]
```diff
@@ -119,13 +122,26 @@ spec:
...
+
+```bash
+kubectl annotate node my-node projectcalico.org/RouteReflectorClusterID=244.0.0.1
 ```
+
+</TabItem>
+<TabItem label="etcd datastore" value="etcd datastore-1">
```
Comment (ctauchen) [inline calico/networking/configuring/bgp.mdx]:
```suggestion
<TabItem label="etcd datastore" value="etcd-datastore">
```

**Review by ctauchen — COMMENTED** [review]
@sridhartigera A few suggestions for the TabItem props. Please make similar changes globally, and make sure to continue the tab structure in CC and CE.

**Comment by sridhartigera** [comment]
> @sridhartigera A few suggestions for the TabItem props. Please make similar changes globally, and make sure to continue the tab structure in CC and CE.

Thanks Chris. Tabs are needed only for OSS. I have made the changes to 3.24 and 3.25 as well.

File: calico-enterprise_versioned_docs/version-3.16/networking/configuring/bgp.mdx [hunk 5]
```diff
@@ -255,7 +255,7 @@ If the default BGP configuration resource does not exist, you need to create it
 
 You can configure an AS for a particular node by modifying the node object using `calicoctl`. For example, the following command changes the node named **node-1** to belong to **AS 64514**.
 
-```
+```bash
 calicoctl patch node node-1 -p '{"spec": {"bgp": {"asNumber": "64514"}}}'
```
Comment (caseydavenport) [inline calico-enterprise_versioned_docs/version-3.16/networking/configuring/bgp.mdx:259]: Maybe not for this PR, but we should find ALL the places calicoctl are still used and replace them with tabs

File: calico-cloud/networking/configuring/bgp.mdx (hunk 1 above)
Comment (caseydavenport) [reply calico-cloud/networking/configuring/bgp.mdx:128]: @sridhartigera this one still needs tabs

File: calico-enterprise/networking/configuring/bgp.mdx [hunk 6]
```diff
@@ -123,13 +123,13 @@ spec:
 
 To configure a node to be a route reflector with cluster ID 244.0.0.1, run the following command.
 
-```
-calicoctl patch node my-node -p '{"spec": {"bgp": {"routeReflectorClusterID": "244.0.0.1"}}}'
+```bash
+kubectl annotate node my-node projectcalico.org/RouteReflectorClusterID=244.0.0.1
```
Comment (caseydavenport) [inline calico-enterprise/networking/configuring/bgp.mdx:127]: This one needs tabs

File: calico-enterprise_versioned_docs/version-3.16/networking/configuring/bgp.mdx [hunk 7]
```diff
@@ -123,13 +123,13 @@ spec:
 
 To configure a node to be a route reflector with cluster ID 244.0.0.1, run the following command.
 
-```
-calicoctl patch node my-node -p '{"spec": {"bgp": {"routeReflectorClusterID": "244.0.0.1"}}}'
+```bash
+kubectl annotate node my-node projectcalico.org/RouteReflectorClusterID=244.0.0.1
```
Comment (caseydavenport) [inline calico-enterprise_versioned_docs/version-3.16/networking/configuring/bgp.mdx:127]: This one needs tabs

**Review by caseydavenport — DISMISSED** [review]

File: calico-cloud/networking/configuring/bgp.mdx (hunk 1 above)
Comment (bmckercher123) [reply calico-cloud/networking/configuring/bgp.mdx:128]: Calico Cloud should not have tabs. It does not support calicoctl, so should be kubectl only (like CE).

Comment (caseydavenport) [reply calico-cloud/networking/configuring/bgp.mdx:128]: Yep. realized too late that these were for enterprise / cloud

**Review by bmckercher123 — DISMISSED** [review]
LGTM

**Review by bmckercher123 — DISMISSED** [review]

**Review by bmckercher123 — APPROVED** [review]

**Pull request merged**